### PR TITLE
Add lint workflow with iglulint job

### LINKED ISSUE #6052: Use GitHub action to validate Snowplow schemas with igluctl lint
We can probably use [iglu-schema-lint from the action marketplace](https://github.com/marketplace/actions/iglu-schema-lint) for this.

### PR DESCRIPTION
**Pull request by WillDaSilva**
Closes #6052

### CODE REVIEW

**Comment by WillDaSilva** [comment]
@pandemicsyn I understand the reasoning behind the new on-disk schema file layout, but why are they in a directory named "iglu-client-embedded"?

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Lint
+
+on:
+  pull_request: {}
+  push:
+    branches: [main]
```
Comment (pandemicsyn) [inline .github/workflows/lint.yml:6]: I don't think we really need to run this workflow on every change, wdyt about limiting this to run only when a schema file actually changes?

**Comment by pandemicsyn** [comment]
@WillDaSilva just incase you haven't come across it yet (https://github.com/snowplow-incubator/snowplow-micro#quick-start)

File: .github/workflows/lint.yml (hunk 1 above)
Comment (aaronsteers) [reply .github/workflows/lint.yml:6]: @pandemicsyn - Agreed. 👍

**Review by aaronsteers — APPROVED** [review]

**Review by edgarrmondragon — APPROVED** [review]

**Review by pandemicsyn — APPROVED** [review]
:shipit:

**Pull request merged**